Mathlib/Algebra/Group/AddChar.lean

```diff
@@ -31,170 +30,248 @@ see `Mathlib.NumberTheory.LegendreSymbol.AddCharacter`.
 additive character
 -/
 
-
-universe u v
-
 /-!
 ### Definitions related to and results on additive characters
 -/
 
+open Multiplicative
+
 section AddCharDef
 
 -- The domain of our additive characters
-variable (A : Type u) [AddMonoid A]
+variable (A : Type*) [AddMonoid A]
 
 -- The target
-variable (M : Type v) [CommMonoid M]
-
-/-- Define `AddChar A M` as `(Multiplicative A) →* M`.
-The definition works for an additive monoid `A` and a monoid `M`,
-but we will restrict to the case that both are commutative rings for most applications.
-The trivial additive character (sending everything to `1`) is `(1 : AddChar A M).` -/
-def AddChar : Type max u v :=
-  Multiplicative A →* M
+variable (M : Type*) [Monoid M]
+
+/-- `AddChar A M` is the type of maps `A → M`, for `A` an additive monoid and `M` a multiplicative
+monoid, which intertwine addition in `A` with multiplication in `M`.
+
+We only put the typeclasses needed for the definition, although in practice we are usually
+interested in much more specific cases (e.g. when `A` is a group and `M` a commutative ring).
+ -/
+structure AddChar where
+  /-- The underlying function.
+  Do not use this function directly. Instead use the coercion coming from the `FunLike`
```

Comment:
Should an empty line be added here (and similarly below), to get a separate paragraph in the docstring as shown on hover or in the HTML docs?

Reply:
There was originally, but the linter flags it as a formatting error.